### PR TITLE
improve performance of navigateUrl action

### PR DESCRIPTION
**Pull request by anserwaseem**
## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/runtime/hooks/useNavigateUrl.ts [hunk 1]
```diff
@@ -1,45 +1,77 @@
...
+          action.inputs,
+          defaultScreenModel,
+          context,
+        );
+      }
 
-    setUrlNavigated(true);
-    return navigate(evaluatedInputs.url, { state: evaluatedInputs.inputs });
-  }, [urlNavigated, evaluatedInputs]);
+      navigate(evaluatedUrl, navigationOptions);
```
Comment (evshi) [inline packages/runtime/src/runtime/hooks/useNavigateUrl.ts:70]: Let's reuse `navigateUrl` here

File: packages/framework/src/hooks/useCommandCallback.ts [hunk 2]
```diff
@@ -163,6 +163,7 @@ export const useCommandCallback = <
             },
             styles: theme.Styles,
             ...mapKeys(theme.Tokens ?? {}, (_, key) => key.toLowerCase()),
+            screenContextModel: screenContext.model,
```
[the remark this answers is not in the document]
Comment (anserwaseem) [reply packages/framework/src/hooks/useCommandCallback.ts:166]: `defaultScreenContext` has the model as undefined, we need `global` and `importedScripts` block from the model

[the remark this answers is not in the document]
Comment (evshi) [reply packages/framework/src/hooks/useCommandCallback.ts:166]: Ah then just use `useScreenModel` where it's needed then. It shouldn't cause any re-renders.

File: packages/framework/src/shared/common.ts [hunk 3]
```diff
@@ -159,3 +159,40 @@ export const error = (value: unknown): void => {
   // eslint-disable-next-line no-console
   console.error(value);
 };
+
+export const expressionReplacer =
```
Comment (evshi) [inline packages/framework/src/shared/common.ts:163]: nit: rename to `replace`

File: packages/framework/src/shared/common.ts [hunk 4]
```diff
@@ -159,3 +159,40 @@ export const error = (value: unknown): void => {
...
+  (evaluate: (expr: string) => string) =>
```
Comment (evshi) [inline packages/framework/src/shared/common.ts:164]: this can be named to `replacer`

**Pull request merged**